weave/flow/eval.py

```diff
@@ -119,7 +119,7 @@ def function_to_evaluate(question: str):
     ```
     """
 
-    dataset: Union[Dataset, list]
+    dataset: Union[Dataset, list[dict]]
```

Comment:
This is already enforced implicitly in the post_init hook since the construction of a `Dataset` enforces this list of dict constraint